guides/attachvolume.mdx

```diff
@@ -0,0 +1,232 @@
+
+```json Response
+{
+  "id": "df5c952c-0567-4140-94d6-59cee96f8caa"
+}
+```
+
+Save the `id` field as you will need it in the next step.
+
+---
+
+### Step 3: Create an Instance with the Volume Attached
+
+To attach the volume, pass its `id` in the `volume_ids` field while creating the instance using the [`/instances/create`](https://docs.shadeform.ai/api-reference/instances/instances-create) endpoint. 
+
+<Note> "volume_ids" must be an array of size one. This is because we only allow attaching one volume to one instance right now. Attaching multiple volumes to a single instance is a feature we will add in the future. </Note>
+
+<CodeGroup>
+  ```bash CLI
+  # Create an instance with the volume attached
+  curl --location 'https://api.shadeform.ai/v1/instances/create' \
+  --header 'x-api-key: <api-key>' \
+  --header 'Content-Type: application/json' \
+  --data '{
+    "cloud": "digitalocean",
+    "region": "tor1",
+    "shade_instance_type": "H100_sxm5",
+    "shade_cloud": true,
+    "name": "the-super-cool-digitalocean-server-volume",
+    "volume_ids": ["b1f2c5d3-29ab-4e7b-9f7e-2c35a1e34d4e"] # be sure to include this line
+  }'
+  ```
+</CodeGroup>
+
+**Example Response**
+
+The response will include the `id` for the instance and a `cloud_assigned_id`. Save this for further operations.
+
+```json Response
+{
+  "id": "cc9f6b74-9825-4854-9e9c-dd50c7e97c3a",
+  "cloud_assigned_id": "720f2a6a-e4ee-488a-ade1-e7892f5d730a"
+}
+```
+
+---
+
+### Step 4: Verify the Volume on the Instance
+
+Once the instance is active, SSH into it to verify that the volume has been mounted. Learn how to SSH into a Shadeform instance using [this guide](https://docs.shadeform.ai/getting-started/quickstart#sshing-into-the-instance).
+
+After connecting, run the following command:
+
+```bash
+# List all block devices attached to the instance
+lsblk
+```
+
+Your output should look something similar to this:
+
+```bash
+NAME    MAJ:MIN RM   SIZE RO TYPE MOUNTPOINTS
+loop0     7:0    0  63.9M  1 loop /snap/core20/2318
+loop1     7:1    0    87M  1 loop /snap/lxd/28373
+loop2     7:2    0  38.8M  1 loop /snap/snapd/21759
+loop3     7:3    0  44.3M  1 loop /snap/snapd/23258
+loop4     7:4    0  63.7M  1 loop /snap/core20/2434
+vda     252:0    0   128G  0 disk 
+├─vda1  252:1    0 127.9G  0 part /
+├─vda14 252:14   0     4M  0 part 
+└─vda15 252:15   0   106M  0 part /boot/efi
+vdb     252:16   0    50G  0 disk # this is the volume I mounted
+nvme0n1 259:0    0 894.3G  0 disk 
+```
+
+This command will display all block devices attached to the instance. You should see your volume listed as a new device, such as `vdb`, `sbd`, or similar.
+
+---
+
+### Step 5: Mount the volume to a file system
+
+After verifying the volume is attached, you can mount it to a file system. Find your volume from the `lsblk` command (my volume is called `vdb` in this example) and mount it to a directory using the following commands:
+
+```bash
+sudo mkdir /dev/vdb
```

Comment:
We don't actually create /dev/vdb. It's not a real folder but a file system representation of a device. This is something that linux does for devices.